hw6/task1/library.js

```diff
@@ -0,0 +1,194 @@
+                                    }                                 
+                                }
+                                if(args[i].constructor.name === "Track"){
+                                        element.traks.push(args[i]);
+                                }
+                                if(args[i].constructor.name === "Album"){
+                                    element.albums.push(args[i]);
+                                }
+                                                         
+                            }
+                        } 
+                    });
+                }                           
+            };         
+    };        
+
+// DELETE METHOD 
+    delete(value){
+                    //TRACK
+        if(value.constructor.name === "Track"){
+           let index = this.Track.indexOf(value);
+           this.Track.splice(index,1);               
+        };
+                    //ALBUM
+        if(value.constructor.name === "Album"){
+            let index = this.Album.indexOf(value);
+            this.Album.splice(index,1);               
+         };
+                    //ARTIST
+        if(value.constructor.name === "Artist"){
+            let index = this.Artist.indexOf(value);
+            this.Artist.splice(index,1);               
+         };         
+    };
+
+// UPDATE METHOD
+    update(myElement, argument, value ){
+                    //TRACK
+        if(myElement.constructor.name === "Track"){
+            this.Track.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }
+                    //ALBUM
+        if(myElement.constructor.name === "Album"){
+            this.Album.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }
+        if(myElement.constructor.name === "Artist"){
+            this.Artist.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }        
+    }
+    
+//PLAY METHOD
+    play(){        
+        const player = new EventEmitter();              
+        let current = 0; 
+        let myTracks = this.Track;
+        let strInfo = `${myTracks[current].name}, Duration: ${myTracks[current].seconds}`; 
+        //EventEmitter next()        
+        this.nextTrack = function(){player.once('next', ()=>{
+            if(current + 1 >= myTracks.length){ 
+                current = 0;                
+            }
+            else{
+                current++;                            
+            }   
+                       
+        })};
+        //EventEmitter prev()    
+        this.prevTrack = function(){player.once('prev', ()=>{
```

Comment:
events will be works only 1 time. It's mean that when I calls prevTrack 2 times a row - only 1 command will be executed

Reply:
fixed